bindings/java/src/main/java/org/apache/opendal/NativeLayer.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal;
+
+public abstract class NativeLayer {
```

Comment:
Can we make `NativeLayer` private and only used inside opendal package?

Reply:
The `op.layer` method accept `NativeLayer` param. I tend to leave it as `public`.

But yes I have the same concern that some public classes or methods are limited to Java's visibility system. I tend to investigate two possibility to improve the situation:

1. Use Java's module system. But it requires JDK 9+, while it doesn't cause an error in JDK 1.8.
2. Use Apache Yetus' [interface audience](https://yetus.apache.org/documentation/in-progress/javadocs/org/apache/yetus/audience/package-summary.html) and other tools to maintain the interface stability and visibility module.